sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -649,6 +659,36 @@ ze_result_t ZeCall::doCall(ze_result_t ZeResult, const char *ZeName,
   if (!(condition))                                                            \
     return error;
 
+pi_result _pi_queue::appendWaitAndResetIfLastEventDiscarded(
+    pi_command_list_ptr_t CommandList) {
+  if (LastCommandEvent && LastCommandEvent->IsDiscarded) {
```

Comment:
What if the last command's event is not discarded but previous one is?

Reply:
If the previous event was discarded it was reset and put into the cache after corresponding command submission.
If the last command is not discarded then it is a new event (we don't take it from cache) and don't reset it after command submission and don't put it to the cache.

Reply:
sounds ok